static/js/components/SearchTextbox.js

```diff
@@ -8,24 +8,49 @@ type Props = {
   value: string
 }
 
-const SearchTextbox = ({ onChange, onClear, onSubmit, value }: Props) => (
-  <div className="search-textbox">
-    <i className="material-icons search-icon" onClick={onSubmit}>
-      search
-    </i>
-    {value ? (
-      <i className="material-icons clear-icon" onClick={onClear}>
-        clear
-      </i>
-    ) : null}
-    <input
-      type="text"
-      className="underlined"
-      value={value}
-      onChange={onChange}
-      onKeyDown={event => (event.key === "Enter" ? onSubmit(event) : null)}
-    />
-  </div>
-)
+export default class SearchTextbox extends React.Component<Props> {
+  input: { current: null | React$ElementRef<typeof HTMLInputElement> }
 
-export default SearchTextbox
+  constructor(props: Props) {
+    super(props)
+    this.input = React.createRef()
+  }
+
+  focus() {
+    if (this.input.current) {
+      this.input.current.focus()
+    }
+  }
+
+  componentDidMount() {
+    this.focus()
+  }
+
+  componentDidUpdate() {
+    this.focus()
```

Comment:
I noticed that one small side-effect of calling this `focus` method in `componentDidUpdate` is that any update to a prop that this component receives will cause the search input to grab focus for a second. this can be a bit odd, especially because clicking on the dropdown to change how the search results are sorted will cause that to happen.

I think on mobile this may cause the keyboard to come up when it happens, but I'm not sure (I don't have access to a simulator to test it out).

Reply:
I modified `componentDidUpdate` to focus only if the input value has changed and is empty.